src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java

```diff
@@ -34,6 +35,14 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         }
         Long id = (Long) req.getSession().getAttribute(DRIVER_ID);
         if (id == null) {
+            req.setAttribute("errorMsg", "Access denied! Log in or register to get access");
+            resp.sendRedirect("/login");
+            return;
+        }
+        try {
+            driverService.get(id);
+        } catch (NoSuchElementException e) {
+            req.setAttribute("errorMsg", "Access denied! Log in or register to get access");
```

Comment:
good. but as for me, it's better not to catch runtime exceptions, but to properly process Optional in service: we can invoke `get` on optional with a check on `Presence` and if no value present - return null. I understand that returning null is a bad practice, but it's better not to show to user stacktrace of an exception as well as not to catch runtime exceptions.

Reply:
I will wait for mentors answer about this.
All in all I agree with you but: 
a) Yes, I can do in service .orElseGet(null) and meaning of Optional will be only in cautioning user that there could be null.
b) When I started to changing this service I got problem that I need to change everything to return Optional and do the same thing as we done in .findByLogin(). It is a right handling of null since I don't give null or exception, I give Optional for user to decide what they want to do with it. But my PR will look like garbage if I do it, so limiting myself with task requirements I left it as is and managed to work with what I have from perspective if I am user of this service.